sw/device/silicon_creator/lib/shutdown.c

```diff
@@ -269,12 +323,38 @@ SHUTDOWN_FUNC(NO_MODIFIERS, shutdown_report_error(rom_error_t reason)) {
   // that we won't jump to a different function.
   uint32_t redacted_error = shutdown_redact_inline(reason, policy);
 
-  // TODO(lowrisc/opentitan#7894): What (if anything) should we print at
-  // startup?
-  log_printf("lc_state: 0x%x\n\r", (unsigned int)raw_state);
+  // Reset UART TX fifo and enable TX.
+  abs_mmio_write32(kUartBase + UART_FIFO_CTRL_REG_OFFSET,
+                   bitfield_bit32_write(0, UART_FIFO_CTRL_TXRST_BIT, true));
+  uint32_t uart_ctrl_reg = abs_mmio_read32(kUartBase + UART_CTRL_REG_OFFSET);
+  uart_ctrl_reg = bitfield_bit32_write(uart_ctrl_reg, UART_CTRL_TX_BIT, true);
+  abs_mmio_write32(kUartBase + UART_CTRL_REG_OFFSET, uart_ctrl_reg);
+
+  // Encodes a 4 character prefix as a 32-bit value. The resulting prefix is the
+  // concatenation of the given characters and '_'.
+#define MSG_PREFIX(a_, b_, c_) ('_' << 24 | (c_) << 16 | (b_) << 8 | (a_))
 
-  // TODO(lowRISC/opentitan#8236): log_printf is in the .text section.
-  log_printf("boot_fault: 0x%x\n", (unsigned int)redacted_error);
+  // Print the error message and the raw life cycle state as reported by the
+  // hardware.
+  shutdown_print(MSG_PREFIX('B', 'F', 'V'), redacted_error);
```

Comment:
Nice!. One more nit: It would be nice to have these log codes in a central location. Maybe a header file?

Reply:
Done